earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -114,15 +114,62 @@ namespace AssetFactory
   {
     using Impl = typename MutableDerivedAssetHandleType::Impl;
     using AssetStorageType = typename Impl::Base;
-    return MutableDerivedAssetHandleType(std::make_shared<Impl>
-                                            (AssetStorageType::MakeStorage(
-                                        ref_, Impl::EXPECTED_TYPE, 
+    return MutableDerivedAssetHandleType(std::make_shared<Impl
+                                        (AssetStorageType::MakeStorage(
+                                        ref_, Impl::EXPECTED_TYPE,
                                         Impl::EXPECTED_SUBTYPE,
                                         inputs_, meta),
                                         config));
   }
 
   template<class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          AssetDefs::Type type_,
+                                          const khMetaData& meta_,
+                                          const ConfigType& config_)
+  {
+      MutableDerivedAssetHandleType asset = Find<MutableDerivedAssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          asset->Modify(meta_, config_);
+          return asset;
+      }
+      return Make<MutableDerivedAssetHandleType, ConfigType>(ref_, type_, meta_, config_);
+  }
+
+  template<class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          AssetDefs::Type type_,
+                                          const std::vector<SharedString>& inputs_,
+                                          const khMetaData& meta_,
+                                          const ConfigType& config_)
+  {
+      MutableDerivedAssetHandleType asset = Find<MutableDerivedAssetHandleType>(ref_, type_);
+      if (asset)
+      {
+          asset->Modify(inputs_, meta_, config_);
+          return asset;
+      }
+      return Make<MutableDerivedAssetHandleType, ConfigType>(ref_, type_, inputs_, meta_, config_);
+  }
+
+  template <class MutableDerivedAssetHandleType, class ConfigType>
+  MutableDerivedAssetHandleType FindMake(const std::string& ref_,
+                                          const std::vector<SharedString>& inputs_,
+                                          const khMetaData& meta_,
+                                          const ConfigType& config_)
+  {
+     MutableDerivedAssetHandleType asset = Find<MutableDerivedAssetHandleType>(ref_);
+     if (asset)
+     {
+         asset->Modify(inputs_, meta_, config_);
+         return asset;
+     }
+     return Make<MutableDerivedAssetHandleType, ConfigType> (ref_, inputs_, meta_, config_);
+  }
+
+  template<class MutableDerivedAssetHandleType, class ConfigType>
+
```

Comment:
Minor: would be nice to get rid of this empty line. I think it's easier to read if the template statement is right next to the function it applies to.